packages/rpc-core/src/rpc-methods/getTokenAccountsByDelegate.ts

```diff
@@ -0,0 +1,136 @@
+}>;
+
+type TokenAccountInfoWithDefaultData = Readonly<{
+    data: Base58EncodedBytes;
+}>;
+
+type TokenAccountInfoWithBase58EncodedData_DEPRECATED = Readonly<{
+    data: Base58EncodedDataResponse;
+}>;
+
+type TokenAccountInfoWithBase64EncodedData = Readonly<{
+    data: Base64EncodedDataResponse;
+}>;
+
+type TokenAccountInfoWithBase64EncodedZStdCompressedData = Readonly<{
+    data: Base64EncodedZStdCompressedDataResponse;
+}>;
+
+type TokenAccountInfoWithJsonData = Readonly<{
+    data: Readonly<{
+        /** Name of the program that owns this account. */
+        program: {
+            info: TokenAccount;
+            type: 'account';
+        };
+        parsed: unknown;
+        space: U64UnsafeBeyond2Pow53Minus1;
+    }>;
+}>;
+
+type TokenAccountWithPubkey<TAccount extends TokenAccountInfoBase> = Readonly<{
+    account: TAccount;
+    pubkey: Base58EncodedAddress;
+}>;
+
+type MintFilter = Readonly<{
+    /** Pubkey of the specific token Mint to limit accounts to */
+    mint: Base58EncodedAddress;
+}>;
+
+type ProgramIdFilter = Readonly<{
+    /** Pubkey of the Token program that owns the accounts */
+    programId: Base58EncodedAddress;
+}>;
+
+type AccountsFilter = MintFilter | ProgramIdFilter;
+
+type GetTokenAccountsByDelegateApiCommonConfig = Readonly<{
+    /** @default "finalized" */
+    commitment?: Commitment;
+    /** The minimum slot that the request can be evaluated at */
+    minContextSlot?: Slot;
+}>;
+
+type GetTokenAccountsByDelegateApiSliceableCommonConfig = Readonly<{
+    /** Limit the returned account data */
+    dataSlice?: DataSlice;
+}>;
+export interface GetTokenAccountsByDelegateApi {
+    /**
+     * Returns all SPL Token accounts by approved Delegate.
+     */
+    getTokenAccountsByDelegate(
+        program: Base58EncodedAddress,
+        filter: AccountsFilter,
+        config: GetTokenAccountsByDelegateApiCommonConfig &
+            GetTokenAccountsByDelegateApiSliceableCommonConfig &
+            Readonly<{
+                encoding: 'base64';
+            }>
+    ): RpcResponse<TokenAccountWithPubkey<TokenAccountInfoBase & TokenAccountInfoWithBase64EncodedData>[]>;
+
+    getTokenAccountsByDelegate(
+        program: Base58EncodedAddress,
+        filter: AccountsFilter,
+        config: GetTokenAccountsByDelegateApiCommonConfig &
+            GetTokenAccountsByDelegateApiSliceableCommonConfig &
+            Readonly<{
+                encoding: 'base64+zstd';
+            }>
+    ): RpcResponse<
+        TokenAccountWithPubkey<TokenAccountInfoBase & TokenAccountInfoWithBase64EncodedZStdCompressedData>[]
+    >;
+
+    getTokenAccountsByDelegate(
+        program: Base58EncodedAddress,
+        filter: AccountsFilter,
+        config: GetTokenAccountsByDelegateApiCommonConfig &
+            Readonly<{
+                encoding: 'jsonParsed';
+            }>
+    ): RpcResponse<TokenAccountWithPubkey<TokenAccountInfoBase & TokenAccountInfoWithJsonData>[]>;
```

Comment:
This needs a base64 fallback, right?

> If jsonParsed is requested but a parser cannot be found, the field falls back to base64 encoding, detectable when the data field is type string.

See:

https://github.com/solana-labs/solana-web3.js/blob/958cc6b5ceaa8eddc79d46ed133e2ceac4593476/packages/rpc-core/src/rpc-methods/common.ts#L80-L91

Reply:
I didn't include that because the accounts returned are known to be token accounts owned by a known token program, so AFAIK they will always be parsed. That is dependent on current behaviour of the solana codebase though, so I'm not sure whether relying on that at the type level here is the right thing to do? Happy to add the base64 fallback if you'd prefer or I'm missing something though!

Reply:
Mmmmm. Smart. Got it!